content/blog/setting-up-github-actions-for-v.md

```diff
@@ -0,0 +1,355 @@
+you want. The `on` section defines the triggers for the workflow. Here, we
+trigger the workflow on every push to the `main` branch. We can also trigger
+it on every tag, or on pushes to a pull request.
+
+Next, we define the primary `build` job in the list of jobs. We define an
+array or _matrix_ called `os` with our desired platforms. This goes in the
+`strategy` section of the job. Then we refer to the current os when we specify
+which type of runner to use.
+
+```yaml {tabWidth=2}
+# ...
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+    runs-on: ${{ matrix.os }}
+```
+
+Note that _macos_ and _windows_ runners have lower availability than
+_ubuntu_ runners. So it might be worthwhile to quickly prototype on
+`ubuntu-latest` and then add `macos-latest` and/or `windows-latest` when
+needed.
+
+We can now defined the _steps_ for the `build` job:
+
+```yaml {tabWidth=2}
+# ...
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Install V
+        uses: vlang/setup-v@v1
+        with:
+          check-latest: true
+
+      - name: Checkout ${{ github.event.repository.name }}
+        uses: actions/checkout@v2
+
+      - name: Check if code is formatted
+        run: |
+          v fmt -diff .
+          v fmt -verify .
+
+      - name: Build ${{ github.event.repository.name }}
+        run: v .
+
+      - name: Run Tests
+        run: v test .
+```
+
+First, we use the `vlang/setup-v@1` GitHub Action to install V for our runner.
+We set `check-latest` to `true` to get the latest updates. Leave this out if
+you prefer more stability. Then we clone the current project using the
+`actions/checkout@v2` action.
+
+We can now start checking the codebase to ensure that it passes the minimum
+standards we set. V has an opinionated auto-formatter available called `vfmt`.
+Here is a sample usage:
+
+```bash
+# Format all V files recursively in this directory
+# AND rewrite them
+v fmt -w .
+# ^^^
+# You'll most likely use this all the time from the root of
+# your V projects.
+
+# Display the formatted output of a sample V file to stdout
+v fmt sample.v
+```
+
+You can run `v help fmt` to get more details on all the available options.
+Two more options we have at our disposal are:
+
+1. `v fmt -diff .` which displays the difference between the original and
+   formatted source, if any. Ideally, we don't any difference before we
+   push the commits.
+2. `v fmt -verify .` which reports an error if it finds an unformatted
+   file in the repository.
+
+Note that we do not automatically format the code in the CI because it might
+lead to huge `diff`s later on and a lot of potentially redundant commits. It
+is also easier to have our tests be _read-only_.
+
+This step executes quickly. Next we make sure that we can build our project.
```

Comment:
what step executes quickly?

Reply:
At this point, there was a detour of 2-3 paragraphs that discuss `v fmt`. It would be helpful to name the step explicitly, instead of referring to it with `this`.